x-pack/platform/plugins/shared/cases/common/observables/validators.ts

```diff
@@ -17,7 +17,23 @@ import {
   OBSERVABLE_TYPE_URL,
 } from '../constants';
 
-const DOMAIN_REGEX = /^(?!-)[A-Za-z0-9-]{1,63}(?<!-)\.[A-Za-z]{2,}$/;
+/**
+ * DOMAIN REGEX breakdown:
+ *
+ * (?=.{1,253}$)
+ *    Ensures the total length of the domain name is between 1 and 253 characters.
+ *
+ * (([a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)\.)+
+ *    Matches one or more labels separated by dots. Each label:
+ *      - Begins and ends with an alphanumeric character (`[a-zA-Z0-9]`).
+ *      - Can contain hyphens (`-`) but not consecutively or at the start/end.
+ *      - Has a maximum length of 63 characters.
+ *
+ * [a-zA-Z]{2,63}
```

Comment:
Maybe an edge case, but URLs can end with dots 😅 I saw a similar issue in a different PR some time ago.

Reply:
Did you mean URLs or domains? 😄 
Just double checking as this is only for domain validation ☺️

Reply:
Look at that (Source: https://datatracker.ietf.org/doc/html/rfc1034#:~:text=When%20a%20user,ISI.EDU%20domain.)
<img width="749" alt="image" src="https://github.com/user-attachments/assets/562e2229-64b4-4026-8e93-c13ab2b2d73e" />